packages/patternfly-4/react-core/src/components/Tabs/Tabs.js

```diff
@@ -14,29 +15,77 @@ const propTypes = {
   className: PropTypes.string,
   /** the index of the active tab */
   activeKey: PropTypes.number,
-  /** handel tab selection */
+  /** handle tab selection */
   onSelect: PropTypes.func,
+  /** uniquely identifies the Tabs */
+  id: PropTypes.string,
   /** Enables the filled tab list layout */
   isFilled: PropTypes.bool,
   /** Enables Secondary Tab styling */
   isSecondary: PropTypes.bool,
   /** Aria Label for the left Scroll Button */
   leftScrollAriaLabel: PropTypes.string,
   /** Aria Label for the right Scroll Button */
-  rightScrollAriaLabel: PropTypes.string
+  rightScrollAriaLabel: PropTypes.string,
+  /** provides an accessible label for the Tabs and defines the Tabs inside a navigation region (labels should be unique for each set of Tabs that are present on a page) */
+  'aria-label': PropTypes.string
 };
 
 const defaultProps = {
   className: '',
   activeKey: 0,
   onSelect: () => undefined,
+  id: null,
   isFilled: false,
   isSecondary: false,
   leftScrollAriaLabel: 'Scroll left',
-  rightScrollAriaLabel: 'Scroll Right'
+  rightScrollAriaLabel: 'Scroll Right',
+  'aria-label': null
 };
 
-class Tabs extends React.Component {
+const Tabs = ({ 'aria-label': ariaLabel, id, ...props }) => {
+  const unique_id = id ? id : getUniqueId();
+  return ariaLabel
```

Comment:
I still don't think you should be switching with the aria-label.  I think you should expose the API for the TabsWithNav and TabsWithDiv and let the consumer decide what to use and compose it themselves.

Reply:
My mistake - updated in new commit.

Reply:
:( I think this is a bit too clunky. Would be much better to use a variant prop then if you don't want to key off of the ariaLabel.

Reply:
What I like about how it is now is that it's  clear that aria-label would only be used for the `<nav>` element because this prop is only included for the TabsWithNav component.

If we do have a single component, with one prop for specifying element and another for specifying `aria-label` then should we update the `aria-label` description to be more descriptive about this relationship? For example, assuming the prop for specifying html element is named `parentElement`:
> Provides an accessible label for the Tabs. Labels should be unique for each set of Tabs that are present on a page. When `parentElement` is set to `nav`, this prop should be defined to differentiate the tabs from other navigation regions on the page.

@jschuler Were you thinking the consumer should be able to pass whatever they want in for the variant prop (e.g. not just `<div>` vs `<nav>` but whatever)? 

I'm still a little fuzzy on what the expectations are related to documentation between react and html examples, like whether we expect react documentation to stand on it's own, or if we expect react devs to also refer to core documentation to better understand how to define some of these props. I'm ultimately fine with whatever approach you've been taking with other components.